src/main/kotlin/dk/cachet/carp/webservices/data/domain/CawsMutableDataStreamBatch.kt

```diff
@@ -0,0 +1,94 @@
+package dk.cachet.carp.webservices.data.domain
+
+import dk.cachet.carp.common.application.data.Data
+import dk.cachet.carp.data.application.*
+import dk.cachet.carp.data.application.DataStreamId
+import dk.cachet.carp.data.application.DataStreamSequence
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+
+/** CHECK create a function that checks sequence IDs before appending
+ * + usable if we can get back last preceding upload of a sequence so the client know what to send
+ * This a "trick" to make the class `DataStreamBatch` open for allowing appendDataStream without checking
+ * all the sequence IDs in the append method. This is a temporary solution until we find a better way to handle this.
+ */
+class CawsMutableDataStreamBatch : Sequence<DataStreamPoint<*>>, DataStreamBatch {
+    companion object {
+        private val LOGGER: Logger = LogManager.getLogger()
+    }
+
+    private val sequenceMap: MutableMap<DataStreamId, MutableList<MutableDataStreamSequence<*>>> = mutableMapOf()
+
+    /**
+     * A list of sequences covering all sequences so far appended to [CawsMutableDataStreamBatch].
+     * This may return less sequences than originally appended in case appended sequences were merged with prior ones.
+     */
+    override val sequences: Sequence<DataStreamSequence<*>>
+        get() = sequenceMap.asSequence().flatMap { it.value }.map { it as DataStreamSequence<*> }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun <TData : Data> appendSequenceInternal(sequence: DataStreamSequence<TData>) {
```

Comment:
- Since _appendSequenceInternal_ is appending the data, and _appendSequence_ is doing a type check, would it make more sense to name them as _appendSequence_, _sequenceTypeCheck_ respectively, and perhaps make the later one a Boolean function then use _require_ for condition check.

Reply:
- Also, so far I see no other implemented use-cases for the both methods, could they be merged? Since as stated in commends it's 'temporary '.

Reply:
- I see no usage for generic type `<TData : Data>`, assuming we only deal with `MutableDataStreamSequence<Data>`, or `DataStreamSequence`, should it be a wildcard argument type (or type projection if you speaking Kotlin) ?

Reply:
I wouldn't merge them, so the codebase is clear and easier to work with later. no need for greater function IF the functions have logical meaning.